system/security/LdapSecurity/ldapconnection.cpp

```diff
@@ -1019,12 +1022,27 @@ class CLdapConnection : implements ILdapConnection, public CInterface
 
     virtual bool validate()
     {
+        if(!m_connected)
+            return connect();
+
+        //Ensure we are not using a rejected LDAP AD host
+        StringBuffer hostbuf;
+        m_ldapconfig->getLdapHost(hostbuf);
+        if (strcmp(hostbuf.str(), m_connectedHost.str()))
+        {
+            if (m_ld)//different host, disconnect from previous and try to reconnect to new one
```

Comment:
reuse: Could have put this code in an disconnect() function, and called from here and later in this function.